creator/data_templates/mutations/template_version.py

```diff
@@ -277,6 +285,64 @@ def mutate(self, info, id):
         return DeleteTemplateVersionMutation(success=True, id=id)
 
 
+class UploadFieldDefinitionsMutation(graphene.Mutation):
+    """Upload a template_version's field definitions from file"""
+
+    class Arguments:
+        file = Upload(
+            required=True,
+            description="The file containing template field definitions",
+        )
+
+    success = graphene.Boolean()
+    file_name = graphene.String()
+    file_size = graphene.Float()
+    field_definitions = graphene.JSONString()
+
+    def mutate(self, info, file):
+        """
+        Imports field definitions from a file (tsv, csv, xlsx, or json):
+
+        - Parse file content into JSON
+        - Best-effort clean up of human errors
+        - Validate JSON against field definitions schema
+
+        Returns the clean version of the field definitions as a JSON string
+        or raises a ValidationError with a list of all errors found
+        """
+        user = info.context.user
+        if not user.has_perm("data_templates.change_datatemplate"):
+            raise GraphQLError("Not allowed")
+
+        file_name = file.name
+        file_size = file.size
+
+        if file.size > settings.FILE_MAX_SIZE:
+            raise GraphQLError(
+                f"File size {file_size} exceeds max size: "
+                f"{settings.FILE_MAX_SIZE}"
+            )
+
+        try:
+            field_definitions = TemplateVersion.load_field_definitions(
+                file, original_name=file.name
+            )
+        except ValidationError:
```

Comment:
Why do you need to specifically filter out `ValidationError` instead of letting it be handled by the catch all below?

Reply:
The catch-all replaces the original exception message with a general one so the user wouldn't get the details of the validation error. I don't think I want the user to see details of every exception so that's why I logged the exception in the catch-all so the stack trace and exception message appear in the app logs, while the the GraphQLError just returns a simple something-went-wrong message to the user on the frontend